lib/smooth/include/smooth/core/network/Wifi.h

```diff
@@ -73,6 +73,8 @@ namespace smooth::core::network
                                             void* event_data);
 
             [[nodiscard]] std::string get_mac_address() const;
+            esp_err_t get_local_mac_address(uint8_t m[6]) const;
+            ip4_addr_t get_local_ip() const;
```

Comment:
Mark these with [[nodiscard]]
The `m` parameter to `get_local_mac_address` is meant to be an outparameter, but since it is not a reference, does it really work? Oh, it decays to a uint8_t*, doesn't it?

Reply:
yes actually it is uint8_t*
I was hoping that defining it like this makes the compiler check the amount of data written to the array (but probably this is not the case).

Reply:
Well, you could use `std::array` and pass the pointer returned by data() to the idf-function. At least that adds safety upwards and prevents anything other than the correct size from being used.

Reply:
o.k. now using `std::array`.